packages/webapp/src/components/Animals/AddAnimalsFormCard/AnimalSelect.tsx

```diff
@@ -0,0 +1,112 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { Controller, FieldValues, UseControllerProps } from 'react-hook-form';
+import { CreatableSelect } from '../../Form/ReactSelect';
+import { components, MenuProps } from 'react-select';
+import { useTranslation } from 'react-i18next';
+
+export type Option = {
+  label: string;
+  value: string | number;
+};
+
+export type AnimalTypeSelectProps = {
+  typeOptions: Option[];
+  onTypeChange?: (Option: Option | null) => void;
+};
+
+export type AnimalBreedSelectProps = {
+  breedOptions: Option[];
+  showNoTypeSelectedMessage: boolean;
+};
+
+export function AnimalTypeSelect<T extends FieldValues>({
+  name,
+  control,
+  typeOptions,
+  onTypeChange,
+}: AnimalTypeSelectProps & UseControllerProps<T>) {
+  const { t } = useTranslation();
+  return (
+    <Controller
+      name={name}
+      control={control}
+      render={({ field }) => (
+        <CreatableSelect
+          label={t('ADD_ANIMAL.TYPE')}
+          placeholder={t('ADD_ANIMAL.TYPE_PLACEHOLDER')}
+          options={typeOptions}
+          onChange={(option) => {
+            field.onChange(option);
+            onTypeChange?.(option);
+          }}
+        />
+      )}
+    />
+  );
+}
+
+export function AnimalBreedSelect<T extends FieldValues>({
+  name,
+  control,
+  breedOptions,
+  showNoTypeSelectedMessage,
+}: AnimalBreedSelectProps & UseControllerProps<T>) {
+  const { t } = useTranslation();
+  return (
+    <Controller
+      name={name}
+      control={control}
+      render={({ field }) => (
+        <CreatableSelect
+          options={breedOptions}
+          label={t('ADD_ANIMAL.BREED')}
+          placeholder={t('ADD_ANIMAL.BREED_PLACEHOLDER')}
+          //@ts-ignore
+          showNoTypeSelectedMessage={showNoTypeSelectedMessage}
```

Comment:
Had to use ts-ignore here because  `showNoTypeSelectedMessage` is a custom prop not defined in ReactSelect, however ReactSelect will still accept any prop and then pass it to custom components. In this case the custom `AnimalBreedMenu` used for displaying message 'please select a type first'.

For extending the types to accept custom props, they recommend to use module augmentation, like: 
```

declare module 'react-select/dist/declarations/src/Select' {
  export interface Props<
    Option,
    IsMulti extends boolean,
    Group extends GroupBase<Option>
  > {
    showNoTypeSelectedMessage: boolean;
  }
}
```

But this makes it so that the custom props are able to be used in all ReactSelect components and according to this [discussion](https://github.com/JedWatson/react-select/issues/4804#issuecomment-945972022), its not possible to specify it per component.